actionpack/lib/action_dispatch/journey/formatter.rb

```diff
@@ -58,8 +58,8 @@ def extract_parameterized_parts(route, options, recall, parameterize = nil)
             !options.key?(part) || (options[part] || recall[part]).nil?
           } | route.required_parts
 
-          parameterized_parts.delete_if do |bad_key, _|
-            !keys_to_keep.include?(bad_key)
+          (parameterized_parts.symbolize_keys.keys - keys_to_keep.map(&:to_sym)).each do |bad_key|
+            parameterized_parts.delete(bad_key)
```

Comment:
Before this commit we are still deleting keys from `parameterized_parts`, the behavior is just changed so that both hashes have symbol keys, to properly process them.